docs/src/pages/[platform]/connected-components/storage/storage-browser/react.mdx

```diff
@@ -0,0 +1,719 @@
+
+<Example>
+<ExampleCode>
+
+```ts
+export const storage = defineStorage({
+  name: 'myProjectFiles',
+  access: (allow) => ({
+    'public/*': [
+      allow.guest.to(['read']),
+      allow.authenticated.to(['read', 'write', 'delete']),
+    ],
+    'protected/{entity_id}/*': [
+      allow.authenticated.to(['read']),
+      allow.entity('identity').to(['read', 'write', 'delete'])
+    ],
+    'private/{entity_id}/*': [
+      allow.entity('identity').to(['read', 'write', 'delete'])
+    ]
+  })
+});
+```
+</ExampleCode>
+</Example>
+
+While not technically the same as S3 Access Grants, the access rules defined in `defineStorage` are treated like 'locations'. Users will see all of the access paths they are authorized to see and can start browsing the S3 bucket at those paths.
+
+Then in your React code, call `Amplify.configure()` with your `amplify_outputs.json`. If you have some access rules that require a logged in user, like `allow.authenticated`, you can wrap your page in the [`<Authenticator>`](/[platform]/connected-components/authenticator) component to easily add authentication flows to your app. 
+
+<Example>
+<ExampleCode>
+```ts
+import {
+  createAmplifyAuthAdapter,
+  createStorageBrowser,
+} from '@aws-amplify/ui-react-storage/browser';
+import "@aws-amplify/ui-react-storage/styles.css";
+
+import config from './amplify_outputs.json';
+
+Amplify.configure(config);
+
+export const { StorageBrowser } = createStorageBrowser({
+  config: createAmplifyAuthAdapter(),
+});
+```
+</ExampleCode>
+</Example>
+
+### IAM Identity Center and S3 Access Grants
+
+<Example>
+<ExampleCode>
+```ts
+import {
+    createManagedAuthAdapter,
+    createStorageBrowser,
+} from '@aws-amplify/ui-react-storage/browser';
+import '@aws-amplify/ui-react-storage/styles.css';
+
+export const { StorageBrowser } = createStorageBrowser({
+   config: createManagedAuthAdapter({
+    credentialsProvider: async (options?: { forceRefresh?: boolean }) => {
+      // return your credentials object
+      return {
+        accessKeyId: 'my-access-key-id',
+        secretAccessKey: 'my-secret-access-key',
+        sessionToken: 'my-session-token',
+        expiration: new Date(),
+      }
+    },
+    // AWS `region` and `accountId`
+    region: '',
+    accountId: '',
+    // call `onAuthStateChange` when end user auth state changes 
+    // to clear sensitive data from the `StorageBrowser` state
+    registerAuthListener: (onAuthStateChange) => {},
+  })
+});
+```
+</ExampleCode>
+</Example>
+
+### Custom auth
+
+Using custom auth you will need to provide the temporary credentials used to sign the requests to S3.
+
+<Example>
+<ExampleCode>
+```ts
+import { createStorageBrowser } from '@aws-amplify/ui-react-storage/browser';
+import '@aws-amplify/ui-react-storage/styles.css';
+
+export const { StorageBrowser } = createStorageBrowser({
+   config: {
+    region: 'XXX',
+    accountId: 'XXXXXX',
```

Comment:
I'd like to callout the `region` and `accountId` here are different to ones in the managed auth example. They are more specifically:
```
// Default AWS `region` and `accountId` of the S3 buckets.
```